transformations/slangificator/transformation.py

```diff
@@ -0,0 +1,226 @@
+import itertools
+import random
+import spacy
+import os
+
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+def slangifyPoS(
+    token, modified_toks, Dictionary, PoS, probReplace, isCap, ReplPot, ReplMade
+):  # performs transformation similar to all three PoS
+
+    # Check if word is in the corresponding dictionary
+    if token.lemma_ in Dictionary[0]:
+        ReplPot += 1  # increment potential replacements
+
+        repDecision = (
+            random.uniform(0, 1) <= probReplace
+        )  # Randomly decide whether to replace or not
+
+        if repDecision:  # if replacement is made
+            ReplMade += 1
+
+            # Choose a new word for replacement
+            # ind=Slang_Adverbs[0].index(token.lemma_)
+            indAllPosRepl = [
+                i for i, x in enumerate(Dictionary[0]) if x == token.lemma_
+            ]  # all possible replacements
+            indChosenRepl = random.randint(
+                0, len(indAllPosRepl) - 1
+            )  # choose one of the replacements
+            indChosenRepl = indAllPosRepl[indChosenRepl]  # index of that replacement
+
+            if PoS == "Noun":  # Treat plular case for nouns
+                # Take plural or signular form. Note only for nouns
+                if token.tag_ == "NN" or token.tag_ == "NNP":
+                    temp = Dictionary[1][
+                        indChosenRepl
+                    ]  # pick the word chosen for replcacement
+                else:
+                    temp = Dictionary[2][
+                        indChosenRepl
+                    ]  # pick the word chosen for replcacement
+            else:
+                temp = Dictionary[1][
+                    indChosenRepl
+                ]  # pick the word chosen for replcacement
+
+            if isCap:  # Make the fist letter capital if necessary
+                temp = temp[0].upper() + temp[1:]
+
+            modified_toks.append(temp + token.whitespace_)
+
+        else:  # if no replacement is made
+            modified_toks.append(token.text + token.whitespace_)
+
+    else:  # if not in the dictionary
+        modified_toks.append(token.text + token.whitespace_)
+
+
+def slangify(
+    self,
+    text,
+    probReplaceNoun=0.5,
+    probReplaceAdjective=0.5,
+    probReplaceAdverb=0.5,
+    seed=0,
+    max_outputs=1,
+):
+    random.seed(seed)
+
+    nlp = self.nlp  # get an instnace of the tokenizer
+    perturbed_texts = []  # output for all perturbed texts
+
+    # Load dictionaries
+    Slang_Nouns = self.Slang_Nouns
```

Comment:
I won't hold your PR for this but just for future reference you might want to keep all variables non-capitalised : either camelCase or underscore_based.